### PR TITLE
Add department ID to Task order ID

### LINKED ISSUE #18017: Update Task Order code to consider DEPT when merging records
### Current State from Charlie:
We're using that TASK_ORDER_ID field as the primary key and if there are duplicate TASK_ORDER_ID's we "coalesce" by just commas separating all of the BYR_FDU's into one text field. It looks 24ART01073 has different data all across all of the fields, not just BYR_FDU so we will need to rework this if we want it to behave differently. 

### Update we'd like to make: 
- Create field that combines TASK_ORDER_ID field and DEPT field as the primary key.
- Use new field `TASK_ORDER_DEPT_ID` generated in Knack to map that connection - #18043

### Background Context: 
When the Public Works and Transportation departments merged in October, finance deactivated all of the TKs in DEPT 2400 and created these TKs again in 6200 (our new DEPT code). Currently the only TKs showing in our applications are the 2400 TKs that are inactive and we are unable to see the active current state of the TK that was created in 6200. Users would like the ability to see that information!

### PR DESCRIPTION
**Pull request by Charlie-Henry**
fixes https://github.com/cityofaustin/atd-data-tech/issues/18017

When deploying this, I'll also have to make a minor change to the airflow docker command from `python` to `python3`, not sure if there's a way to change that?

Not sure what reviews need to be done, but @ChristinaTremel has verified the data I've loaded into a [test data tracker ](https://builder.knack.com/atd/test--austin-transportation-data-tracker--22-feb-2024/schema/list/objects/object_86/fields)looks good.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -67,6 +71,11 @@ def stringify_value(value):
                 "src": "TASK_ORDER_ID",
                 "data-tracker": "field_1277",
                 "finance-purchasing": "field_989",
+            },
+            {
+                "src": "DEPT_TK_ID",
+                "data-tracker": "field_4837",
+                "finance-purchasing": "field_1113",
                 "primary_key": True,
```
Comment (Charlie-Henry) [inline config.py:79]: New primary key is the Department Code + Task order ID

File: queries.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
     SELECT
         atd_tk.TASK_ORDER_DEPT,
         atd_tk.TASK_ORDER_ID,
+        atd_tk.TASK_ORDER_DEPT || atd_tk.TASK_ORDER_ID as DEPT_TK_ID,
```
Comment (Charlie-Henry) [inline queries.py:13]: Biggest change:
New primary key is created in the SQL query of the finance DB

File: Dockerfile [hunk 3]
```diff
@@ -1,11 +1,8 @@
-FROM atddocker/atd-oracle-py:production
+FROM ghcr.io/oracle/oraclelinux8-instantclient:19
```
Comment (Charlie-Henry) [inline Dockerfile:1]: dropping the old atd-oracle-py which doesn't work on arm64 in favor of using an oracle provided docker image. h/t @frankhereford

File: s3_to_knack.py [hunk 4]
```diff
@@ -220,13 +233,18 @@ def main():
...
+    with ThreadPoolExecutor(max_workers=10) as executor:
+        futures = []
+        for record in todos:
+            futures.append(
+                executor.submit(process_record, record, knack_obj, app, count)
+            )
+            count += 1
+
+    for future in futures:
+        future.result()  # This will re-raise any exceptions encountered in the threads
```
Comment (johnclary) [inline s3_to_knack.py:247]: nice!

[like i mentioned here](https://github.com/cityofaustin/atd-vz-data/pull/1478/files#r1668575664), akaik there's not a clean way to short-circuit this loop until all records are processed. this probably isn't a big deal 🤷

File: Dockerfile (hunk 3 above)
Comment (johnclary) [reply Dockerfile:1]: good stuff. there's probably some way to alias `python=python3`.  but i've never tried this within a docker file. i'd probably just update the Airflow command out of laziness

**Review by johnclary — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢

File: Dockerfile (hunk 3 above)
Comment (mddilley) [reply Dockerfile:1]: +1 - seems possible to add aliasing commands in this file, but it probably take some trial & error. i can review the Airflow PR to help you ship this when you need.

**Review by mddilley — APPROVED** [review]
Looks great to me! 🚢

**Pull request merged**